### DISCUSSION TITLE
multi label models &&optimal sample length

### DISCUSSION THREAD

**Discussion by georgemossad** [timeseriesAI/tsai#712] [Q&A] [answered]
Hello! Hope you're all doing well

 have a question to ask: What is the best model to use for analyzing time series data of multi-labeled vibration signal readings in industrial machines, which consist of (4096 points or 2048 points) and what is the optimal sample length to divide the signals?
 
Any insights or suggestions would be greatly appreciated!

**Comment by oguiza** [comment, marked as answer]
Hi @georgemossad ,
There's no way to answer your questions without running any tests. 
The only thing I could recommend is to test a variety of models (always choose the Plus version if available): CNNs, RNNs, Rockets, Transformers, and Wavelet-based. 
If the input and/or output time series is very long, you may have issues with Transformers which use a lot of memory. 
I'd start running tests with a subset of the original dataset so that you can run them quickly.

**Reply by georgemossad** [reply]
thanks for responding,
currently my data is small so I managed to do as you suggested,

but I made a sliding window to split the original data into overlapped samples and then used the get_splits function to get my splits so the accuracy results and losses improved even in less epoch training numbers.

with that, I am afraid that there might be leakage between training, validation, and test samples.

below is my get_slpits function:
'''
n_splits                = 1
valid_size              = 0.2
test_size               = 0.25
train_only              = False  # set to True for debugging (valid = train)
stratify                = True
balance                 = False
shuffle                 = True
predefined_splits       = None
show_plot               = True 

check_splits = True
random_state = 23


splits = get_splits(y, n_splits=n_splits, valid_size=valid_size, test_size=test_size, shuffle=shuffle, balance=balance, stratify=stratify,
                     check_splits=check_splits, random_state=random_state, show_plot=show_plot, verbose=True)
''''

**Reply by oguiza** [reply]
It's difficult to say without knowing the exact details of your task (which is outside the scope of this discussion). You need to define if the train and inference time series will overlap in the real-world (they often partly overlap), so you can use the same approach during traiing.

**Reply by georgemossad** [reply]
thanks for your time, that was very helpful
